lib/bundler/static_setup.rb

```diff
@@ -0,0 +1,43 @@
+require 'bundler/setup'
+
+# cache rack -> /foo/bar/vendor/bundle/rack/lib/rack.rb
+module Bundler
+  PRELOADED_REQUIRE = {}
+  $LOAD_PATH.reverse_each do |path|
+    Dir["#{path}/**/*.rb"].each do |f|
+      start = path.size + 1
+      count = f.size - start - 3 # without '.rb' = last 3
```

Comment:
with just .rb hits 4698 - misses 405
with .so,.o added it's the same ... so seems to not make any difference ...
and even if it does probably not worth the extra complexity/time (search
with pattern + extracting extname) ... on miss it falls back to default
ruby anyway

On Sun, Aug 23, 2015 at 4:05 AM, Josef Šimánek <notifications@github.com>
wrote:

> In lib/bundler/static_setup.rb
> <https://github.com/bundler/bundler/pull/3958#discussion_r37704852>:
>
> > @@ -0,0 +1,43 @@
> > +require 'bundler/setup'
> > +
> > +# cache rack -> /foo/bar/vendor/bundle/rack/lib/rack.rb
> > +module Bundler
> > +  PRELOADED_REQUIRE = {}
> > +  $LOAD_PATH.reverse_each do |path|
> > +    Dir["#{path}/**/*.rb"].each do |f|
> > +      start = path.size + 1
> > +      count = f.size - start - 3 # without '.rb' = last 3
>
> Is this (.rb) really universal?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bundler/bundler/pull/3958/files#r37704852>.
>